File: tests/scenarios/01-apply.lua
-- memory,abt-io
-- This test applies 3 entries to the state machine,
-- performs a barrier, then checks that the content
-- of the state machine is as expected in all workers.

expected = ""
for i=1,3 do
  local entry = string.format("entry_%d", i)
  cluster[i]:apply(entry)
  expected = expected .. entry
end
cluster[1]:barrier()
for i=1,3 do
  content = cluster[i]:get_fsm_content()
  assert(content == expected)
end

File: tests/scenarios/01-apply.py
# This test applies 3 entries to the state machine,
# performs a barrier, then checks that the content
# of the state machine is as expected in all workers.
from raft import cluster
import time

expected = ""
for i in range(0, 3):
    entry = f"entry_{i}"
    cluster[i+1].apply(entry)
    expected = expected + entry

cluster[1].barrier()

time.sleep(1)

for i in range(0,3):
  content = cluster[i+1].get_fsm_content()
  assert content == expected, f"in worker {i+1} content: {content} differs from {expected}"
